core/modules/server/routes/get-index.js

```diff
@@ -17,9 +17,10 @@ exports.method = "GET";
 exports.path = /^\/$/;
 
 exports.handler = function(request,response,state) {
-	var text = state.wiki.renderTiddler(state.server.get("root-render-type"),state.server.get("root-tiddler")),
-		responseHeaders = {
-		"Content-Type": state.server.get("root-serve-type")
+	var text = state.wiki.renderTiddler(state.server.get("root-render-type"),state.server.get("root-tiddler"));
+	var responseHeaders = {
+		"Content-Type": state.server.get("root-serve-type"),
+		"Server-Timing": $tw.perf.generateHeader() || ""
```

Comment:
I am not keen on  the approach of passing performance data back to the client in all responses. It exposes internal information about the server that could be used to attack it. I think it's safer for server logs to stay on the server in the usual way, and only expose them via a privileged admin interface.

Reply:
This measurement header will only be sent when user toggle this feature in the control panel (`if(!this.enabled) return;`), which can only be opened by privileged admin.

And this is [server-timing](https://www.w3.org/TR/server-timing/), which is meant to send to the client. In this way, it is also easy to use.